bcda/auth/provider.go

```diff
@@ -2,14 +2,58 @@ package auth
 
 import (
 	"os"
+	"strings"
 
-	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/dgrijalva/jwt-go"
+	log "github.com/sirupsen/logrus"
 )
 
-// Provider is an interface for operations performed through an authentication provider.
+const (
+	Alpha = "alpha"
+	Okta  = "okta"
+)
+
+var providerName = Alpha
+
+func init () {
+	SetProvider(strings.ToLower(os.Getenv(`BCDA_AUTH_PROVIDER`)))
+}
+
+func SetProvider(name string) {
+	if name != "" {
+		switch (strings.ToLower(name)) {
+		case Okta:
+			providerName = name
+		case Alpha:
+			providerName = name
+		default:
+			log.Infof(`Unknown providerName %s; using %s`, name, providerName)
+		}
+	}
+	log.Infof(`Auth is made possible by %s`, providerName)
```

Comment:
This one prints the correct value, though I notice we're not using the structured logging here. Is there a ticket to add that?

Reply:
BCDA-866

Reply:
and I added the formatting statement to init()